lib/string_reverse.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string.nil?

```

Comment:
Would you want to reverse a string that was whitespace or exit early like you have done so with the `nil` check?